calico_cni/calico_cni.py

```diff
@@ -400,14 +424,17 @@ def _get_policy_driver(self):
         try:
             self.cni_args["K8S_POD_NAME"]
         except KeyError:
-            _log.debug("Using default dolicy driver")
+            _log.debug("Using default policy driver")
             try:
                 driver = policy_drivers.DefaultPolicyDriver(self.network_name)
             except ValueError:
-                _log.error("Invalid characters detected in the network name "
-                           "'%s'. Only letters a-z, numbers 0-9, and _.- "
-                           " are supported", self.network_name)
-                sys.exit(1)
+                self.env["command"] = CNI_CMD_DELETE
+                self._release_ip(self.env)
+                message = "Invalid characters detected in the network name, " \
```

Comment:
I think this message should live in the DefaultPolicyDriver, and this code should just access the `message` attribute on the ValueError